safe_landing_planner/src/nodes/safe_landing_planner_node.cpp

```diff
@@ -222,4 +214,47 @@ void SafeLandingPlannerNode::publishSerialGrid() {
   grid_pub_.publish(grid);
   grid_seq++;
 }
+
+void SafeLandingPlannerNode::pointCloudTransformThread() {
+  while (!should_exit_) {
+    {
+      std::unique_lock<std::mutex> cloud_msg_lock(*(cloud_msg_mutex_));
+      cloud_ready_cv_->wait(cloud_msg_lock);
+    }
+    while (cloud_transformed_ == false) {
+      if (should_exit_) break;
+
+      std::lock_guard<std::mutex> cloud_msg_lock(*(cloud_msg_mutex_));
```

Comment:
There might be an issue here, the lock_guard stays locked even while the `sleep` is happening.

I think a better solution here could be:
```c++
std::unique_lock<std::mutex> cloud_msg_lock(*(cloud_msg_mutex_));
if (cloud_msg_lock.try_lock()) {
    if(tf_listener_.canTransform("/local_origin",
                                    newest_cloud_msg_.header.frame_id,
                                    newest_cloud_msg_.header.stamp)) {
        pcl::PointCloud<pcl::PointXYZ> pcl_cloud;
        // transform message to pcl type
        pcl::fromROSMsg(newest_cloud_msg_, pcl_cloud);
        cloud_msg_lock.unlock();
        .......
    }
    else {
        cloud_msg_lock.unlock();
        ros::Duration(0.001).sleep(); // sleep because transform wasn't ready
    }
}
else {
    ros::Duration(0.001).sleep(); // sleep because cloud_msg_mutex_ was locked
}
```

(@baumanta this for the local_planner too)